src/main/NWN/API/Object/NwCreature.cs

```diff
@@ -1489,5 +1489,37 @@ public async Task SpeakOneLinerConversation(string dialogResRef = "", NwGameObje
       await WaitForObjectContext();
       NWScript.SpeakOneLinerConversation(dialogResRef, tokenTarget);
     }
+
+    /// <summary>
+    /// Fades the screen for a given creature or player from black to regular screen.
+    /// </summary>
+    /// <param name="creature">Creature to fade the screen of from black to normal.</param>
+    /// <param name="fadeSpeed">Determines how fast the fade occurs.</param>
+    public async Task FadeFromBlack(NwCreature creature, float fadeSpeed)
+    {
+      await WaitForObjectContext();
+      NWScript.FadeFromBlack(creature, fadeSpeed);
+    }
+
+    /// <summary>
+    /// Fades the screen for a given player or creature from a regular screen to black.
+    /// </summary>
+    /// <param name="creature">Creature controlled by a player to fade the screen of.</param>
+    /// <param name="fadeSpeed">Determines how fast the fade occurs.</param>
+    public async Task FadeToBlack(NwCreature creature, float fadeSpeed)
+    {
+      await WaitForObjectContext();
+      NWScript.FadeToBlack(creature, fadeSpeed);
+    }
+
+    /// <summary>
+    /// Removes any current fading effects or black screen from the monitor of the player controlling Creature.
+    /// </summary>
+    /// <param name="creature">Creature to fade the screen of from black to normal.</param>
+    public async Task StopFade(NwCreature creature)
+    {
+      await WaitForObjectContext();
+      NWScript.StopFade(creature);
+    }
```

Comment:
The async/await shouldn't be needed here as `creature` is a parameter of the NWScript function.

Reply:
This should probably be moved to NwPlayer too.

Reply:
> This should probably be moved to NwPlayer too.

Yeah, I wasn't sure with description if this worked on anything but PC.

> The async/await shouldn't be needed here as `creature` is a parameter of the NWScript function.

Only reason I added that param is that I was thinking it worked on pc and creature.  If it doesn't work on creature should we bother with the param or keep it and just limit it to NwPlayer?  I'm gonna do that unless you say otherwise.

Reply:
Yep, move it to NwPlayer, and change it to use "this"
```cs
    public async Task StopFade(NwCreature creature)
    {
      await WaitForObjectContext();
      NWScript.StopFade(creature);
    }
    ```
to
```cs
    public void StopFade()
    {
      NWScript.StopFade(this);
    }
```